modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -263,14 +263,22 @@ object Inputs {
     stdinOpt: => Option[Array[Byte]],
     acceptFds: Boolean,
     enableMarkdown: Boolean,
-    isRunWithShebang: Boolean
+    programInvokeData: ScalaCliInvokeData
   ): Seq[Either[String, Seq[Element]]] = args.zipWithIndex.map {
     case (arg, idx) =>
       lazy val path      = os.Path(arg, cwd)
       lazy val dir       = path / os.up
       lazy val subPath   = path.subRelativeTo(dir)
       lazy val stdinOpt0 = stdinOpt
       lazy val content   = os.read.bytes(path)
+      val (isRunWithShebang, isRunWithDefault) = programInvokeData.subCommand match
+        case SubCommand.Shebang => (true, false)
+        case SubCommand.Default => (false, true)
+        case _                  => (false, false)
```

Comment:
```suggestion
      val (isRunWithShebang, isRunWithDefault) =  
          (programInvokeData.subCommand.is(SubCommand.Shebang),
           programInvokeData.subCommand.is(SubCommand.Default))
```
Maybe this approach will be more readable.